src/lib/server/prisma/neovimconfigs/service.ts

```diff
@@ -2,585 +2,605 @@ import { Prisma, type NeovimConfig, type WhereType } from '@prisma/client';
-		include: {
-			user: {
-				select: {
-					githubToken: {
-						select: {
-							accessToken: true
-						}
-					}
-				}
-			}
-		}
-	});
-	return configs
-		.map(({ user, ...config }) => {
-			return {
-				...config,
-				_token: user.githubToken?.accessToken
-			};
-		})
-		.filter((c) => !!c._token) as NeovimConfigWithToken[];
+  const configs = await prismaClient.neovimConfig.findMany({
+    include: {
+      user: {
+        select: {
+          githubToken: {
+            select: {
+              accessToken: true
+            }
+          }
+        }
+      }
+    }
+  });
+  return configs
+    .map(({ user, ...config }) => {
+      return {
+        ...config,
+        _token: user.githubToken?.accessToken
+      };
+    })
+    .filter((c) => !!c._token) as NeovimConfigWithToken[];
 }
 
 export async function getConfigsForPlugin(
-	owner: string,
-	name: string,
-	take: number
+  owner: string,
+  name: string,
+  take: number
 ): Promise<NeovimConfigWithMetaData[]> {
-	const configs = await prismaClient.neovimConfig.findMany({
-		include: {
-			user: {
-				select: {
-					avatarUrl: true
-				}
-			},
-			_count: {
-				select: {
-					neovimConfigPlugins: true
-				}
-			},
-			neovimConfigPlugins: {
-				select: {
-					paths: true
-				},
-				where: {
-					plugin: {
-						owner,
-						name
-					}
-				}
-			}
-		},
-		where: {
-			neovimConfigPlugins: {
-				some: {
-					plugin: {
-						owner,
-						name
-					}
-				}
-			}
-		},
-		orderBy: [
-			{
-				stars: 'desc'
-			},
-			{
-				repo: 'asc'
-			},
-			{
-				root: 'asc'
-			}
-		],
-		take
-	});
-	return configs.map(attachMetaData);
+  const configs = await prismaClient.neovimConfig.findMany({
+    include: {
+      user: {
+        select: {
+          avatarUrl: true
+        }
+      },
+      _count: {
+        select: {
+          neovimConfigPlugins: true
+        }
+      },
+      neovimConfigPlugins: {
+        select: {
+          paths: true
+        },
+        where: {
+          plugin: {
+            owner,
+            name
+          }
+        }
+      }
+    },
+    where: {
+      neovimConfigPlugins: {
+        some: {
+          plugin: {
+            owner,
+            name
+          }
+        }
+      }
+    },
+    orderBy: [
+      {
+        stars: 'desc'
+      },
+      {
+        repo: 'asc'
+      },
+      {
+        root: 'asc'
+      }
+    ],
+    take
+  });
+  return configs.map(attachMetaData);
 }
 
+export async function toggleConfigMarkedForDeletion(id: number, username: string) {
+  const existing = await getConfigs({ user: { username }, id });
+  if (existing.length === 0) {
+    // This means the user doesn't have any such config.
+    return;
+  }
+  console.log(existing[0]);
+  const conf = existing[0]
+  return await prismaClient.neovimConfig.update({
+    where: {
+      id,
+    },
+    data: {
+      markedForDeletion: !conf.markedForDeletion
+    }
+  });
+}
```

Comment:
A couple improvements on this function
* It would be preferable to use userId over username just so it is a little bit more robust.
* This could be simplified by using the userId in the where field in order to filter on the userId instead of having an early check and return, it would error if we cannot find a config
* Early early return would not be ideal here, would prefer a 403 error if this API was called, maybe from the TRPC handle
* We likely can call delete immediately as well, but the function would of course have to be renamed